source/scpd/Cpp.d

```diff
@@ -34,7 +34,72 @@ import std.meta;
 
 import vibe.data.json;
 
+extern(C++) {
+@trusted nothrow @nogc:
+    void defaultCtorCPPObject(T) (T* ptr);
+    void dtorCPPObject(T) (T* ptr);
+    void opAssignCPPObject(T) (T* lhs, T* rhs);
+    void copyCtorCPPObject(T) (T* ptr, inout(T)* rhs);
+    void copyCtorCPPObject(T) (immutable(T)* ptr, inout(T)* rhs);
+    int getCPPSizeof(T) ();
+}
+
+private mixin template CPPBindingMixin (T)
+{
+    extern(D)
+    {
+        this (char dummy) @trusted nothrow @nogc
```

Comment:
Would it be a good idea to add `@disable this();` so people remember to call this ctor?

Reply:
Yes to `@disable this();`

I would also suggest creating a special enum and only using this, e.g. `enum CppCtor { Use = 0; }` and then:
```D
public this (CppCtor use);
auto x = CppType(CppCtor.Use);
```

Reply:
Yep, default ctor was meant to be disabled.